great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -2770,15 +2760,8 @@ def update_expectation_suite(
         """
         expectation_suite_name: str = expectation_suite.expectation_suite_name
         key = ExpectationSuiteIdentifier(expectation_suite_name=expectation_suite_name)
-        if not self.expectations_store.has_key(key):  # noqa: W601
-            raise gx_exceptions.DataContextError(
-                f"expectation_suite with name {expectation_suite_name} does not exist."
-            )
```

Comment:
None of these `has_key` checks should happen in the context - these should happen at the store level